Module_7.2/Q4.js

```diff
@@ -0,0 +1,61 @@
+// Given 2 arrays with related objects, return a new array where objects having the same id from each of the arrays are merged. Try to achieve it with a complexity - O(n).
+
+let userNames = [{
+  "id": 1,
+  "first_name": "Nicki",
+  }, {
+  "id": 2,
+  "first_name": "Raychel",
+  }, {
+  "id": 3,
+  "first_name": "Demetris",
+  }, {
+  "id": 4,
+  "first_name": "Amata",
+  }]
+  let userEmails = [{
+  "id": 2,
+  "email": "rmcgrady1@cpanel.net",
+  }, {
+  "id": 1,
+  "email": "ncrozier0@squarespace.com",
+  }, {
+  "id": 4,
+  "email": "abraiden3@canalblog.com",
+  }, {
+  "id": 3,
+  "email": "dkilshall2@elpais.com",
+  }]
+  
+  // const mergeById = (a1, a2) =>
+  //     a1.map(itm => ({
+  //         ...a2.find((item) => (item.id === itm.id)),
+  //         ...itm
+  //     }));
+  
+  // console.log(mergeById(userNames, userEmails));
+  
+  function mergeById(userNames,userEmails){
+      userNames.sort((a,b)=>a.id-b.id)
```

Comment:
We usually don't update the function arguments directly. Here you are using .sort on the arguments directly.